UI showing Queue list and Contract Store

chatch opened the pull request:
Prototype UI dashboard displaying state of queues, contract store, etc.

Commit:
integrate webpack and add dapp skeleton
builds and loads but web3 not initialised properly

Commit:
Queue and DollarToken now deploy with standard mechanism and register with Services.sol
removed Stub contracts

chatch committed:
Merge pull request #21 from Bakt/deployment_unify

Deployment unify - Queue and DollarToken

Commit:
add WEBAPP.md notes for the webapp
add ipfs deploy command
support non web3 environments using a backup public endpoint for RPC calls
now works with Metamask

Commit:
Merge remote-tracking branch 'origin/dapp' into dapp

chatch committed:
fix backticks and code block

Commit:
merge in latest master

Commit:
support local dev from latest deploy AND deployed testnet from hard wired contract address
use Services to pull deployed addresses
introduce config.json

Commit:
multiple updates to UI - now shows both queues, contract list and accounts
added 'loading ...' notifier
hide kovan server address from config (server under constant attack with ssh login attempts) - so fill it in before deployment but never commit to git

Commit:
move readme to app folder

gnidan commented on app/config.json:
```diff
@@ -0,0 +1,10 @@
+    "network": "development",
+    "testnet": {
+        "services": "0x60b1a811f5ea71806bb3bc0e5b74920eeae91913",
+        "endpoint": "http://?.?.?.?:8545"
```
Since the truffle.js relies on hostnames following the scheme in gnidan/docker-eth-dev, should the dapp part be set up that way as well?

chatch replied:
yes that makes sense. i'll change it to use the truffle.js then.

chatch replied:
@gnidan I've made this change now. See app.js I'm pulling client host from truffle.js. I had to remove the default builder from truffle.js as I get errors on import in app.js. I think it doesn't matter and the default builder gets set anyway? In any case i'm using webpack seperatly to build. Let me know if it all looks okay.

Commit:
merge in WithdrawalReserves changes

gnidan commented:
@chatch I believe this could use the existing truffle service in docker-compose.

I'm going to see about [adding IPFS support](https://github.com/gnidan/docker-eth-dev/issues/11) to my repo and then this should be a non-issue to switch hostnames around.

I'd like to try this out locally and play with it before I'd feel comfortable reviewing

Commit:
Merge branch 'master' into dapp

Commit:
add store and lookup by account to ContractStore
update test to store address of an actual BVC contract along with the accounts involved

Commit:
add BVC details to the contracts view - table now shows emitter, bene and cents values for each contract
render account addresses in short form and add tooltip for seeing the full address

Commit:
use truffle.js defined hosts instead of in config.json
removed builder setting in truffle.js

chatch commented:
@gnidan This is ready for a (re)review now. Can you take a look at latest changes. Major things are:

- updated to use truffle.js host as per your comment above
- updates to contracts view - now shows BVC details in table
- merged in all of latest master branch

gnidan commented on app/javascripts/app.js:
```diff
@@ -0,0 +1,216 @@
+
+import truffleJs from '../../truffle'
+import config from '../config.json'
+const netConfig = truffleJs.networks[config.network]
```
oh, nice!

gnidan commented on truffle.js:
```diff
@@ -1,16 +1,12 @@
         },
         "testnet": {
-            host: "parity-testnet",
+            host: "52.221.123.6",
```
Assuming these got checked in by accident? I don't think those IPs are static at all, or portable. did you have trouble with the hostnames?

chatch replied:
oops. yes i've just changed these back!

gnidan reviewed (changes requested):
Just requesting the truffle.js change, the config.json looks nice! (Although we might have to figure out a dynamic services address later)

gnidan commented:
Other than that, looks good!

Commit:
oops - restore hostname based hosts in truffle.js

gnidan reviewed (approved):
bam!

chatch commented:
thanks mate!
